### PR TITLE
Added support for custom queries

### PR DESCRIPTION
**Pull request by martinslota**
I'll add a few additional comments in the next commit.

### CODE REVIEW

File: .ocamlformat [hunk 1]
```diff
@@ -0,0 +1 @@
+doc-comments=before
```
Comment (Leonidas-from-XIV) [inline .ocamlformat]: I think it would be better to copy the entire configuration to have it consistent, instead of just a single setting.

File: src/errors.ml [hunk 2]
```diff
@@ -66,29 +66,36 @@ module Lookup_ip_error = struct
...
+    [ `Invalid_lookup_path of string
+    | `Invalid_node_number of string
+    | `Unsupported_data_type of string
+    | `Unexpected_data_type of string
+    | Common_error.t ]
   [@@deriving show]
 
+  let of_error_code error_code =
+    let get_error_message () = Error_code.to_message error_code in
+    Mmdb_types.Error_code.(
```
Comment (Leonidas-from-XIV) [inline src/errors.ml]: I think the module open here is rather unclear: What is being included from the module? Being more explicit here would improve understandability, I think.

File: src/mmdb.ml [hunk 3]
```diff
@@ -64,46 +73,48 @@ module Supported_data_type = struct
...
       | _ -> None)
-end
-
-module Query_result = struct
-  type t = [`String of string | `Float of float | `Int of int | `Bool of bool]
 
-  let get_query_result data_type entry_data_ptr =
-    let module T = Supported_data_type in
+  let convert_to_value data_type entry_data_ptr =
     Mmdb_ffi.Helpers.(
```
Comment (Leonidas-from-XIV) [inline src/mmdb.ml]: Similarly here I am very confused what comes from `Helpers`, especially since down below it still refers to e.g. `get_entry_data_size` by full path.

File: src/mmdb.ml [hunk 4]
```diff
@@ -113,74 +124,156 @@ let run_query ~lookup_result ~query :
...
+
+  type answer
+
+  val from_db : t -> Ip.t -> Query.t -> (answer option, Fetch_error.t) Result.t
+
+  val from_ip_data :
+    ip_data -> Query.t -> (answer option, Fetch_value_error.t) Result.t
+end
+
+module Any = struct
```
Comment (Leonidas-from-XIV) [inline src/mmdb.ml]: Would it not be good to constrain this to `VALUE_TYPE` here?

File: src/errors.ml (hunk 2 above)
Comment (martinslota) [reply src/errors.ml]: I get your point but I disagree. Without this the code below gets way too wordy which only obscures the simple value mapping it expresses.

Besides, this is an implementation detail that has little to do with this PR.

File: src/mmdb.ml (hunk 3 above)
Comment (martinslota) [reply src/mmdb.ml]: Similarly as in my other reply.

The full path is a remnant of some older version - I'll remove that.

File: src/mmdb.ml (hunk 4 above)
Comment (martinslota) [reply src/mmdb.ml]: Well, I don't really see any point in doing that...
